test/integration/controller/core/queue_controller_test.go

```diff
@@ -0,0 +1,105 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueue "sigs.k8s.io/kueue/api/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("Queue controller", func() {
+	var (
+		ns                *corev1.Namespace
+		Queue             *kueue.Queue
+		clusterQueue      *kueue.ClusterQueue
+		emptyQueueStatus  kueue.QueueStatus
+		loadedQueueStatus kueue.QueueStatus
+	)
+
+	loadedQueueStatus = kueue.QueueStatus{PendingWorkloads: 3}
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-queue-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.BeforeEach(func() {
+		clusterQueue = testing.MakeClusterQueue("clusterQueue_queue-controller").
+			Resource(testing.MakeResource(resourceGPU).
+				Flavor(testing.MakeFlavor(flavorModelA, "5").Ceiling("10").Obj()).
+				Flavor(testing.MakeFlavor(flavorModelB, "5").Ceiling("10").Obj()).Obj()).Obj()
+		Queue = testing.MakeQueue("queue", ns.Name).
+			Capacity("cluster-queue").Obj()
+		gomega.Expect(k8sClient.Create(ctx, Queue)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteQueue(ctx, k8sClient, Queue)).To(gomega.Succeed())
+		gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should update status when workloads are created", func() {
+		workloads := []*kueue.QueuedWorkload{
+			testing.MakeQueuedWorkload("one", ns.Name).
+				Queue(Queue.Name).
+				Request(corev1.ResourceCPU, "2").Request(resourceGPU, "1").Obj(),
+			testing.MakeQueuedWorkload("two", ns.Name).
+				Queue(Queue.Name).
+				Request(corev1.ResourceCPU, "3").Request(resourceGPU, "1").Obj(),
+			testing.MakeQueuedWorkload("three", ns.Name).
+				Queue(Queue.Name).
+				Request(corev1.ResourceCPU, "1").Request(resourceGPU, "1").Obj(),
+		}
+
+		ginkgo.By("Creating workloads")
+		for _, w := range workloads {
+			gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+		}
+		gomega.Eventually(func() kueue.QueueStatus {
+			var updatedQueue kueue.Queue
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(Queue), &updatedQueue)).To(gomega.Succeed())
+			return updatedQueue.Status
+		}, framework.Timeout, framework.Interval).Should(testing.Equal(loadedQueueStatus))
+
+		ginkgo.By("Finishing workloads")
+		for _, w := range workloads {
+			w.Status.Conditions = append(w.Status.Conditions, kueue.QueuedWorkloadCondition{
```

Comment:
we just need to "admit" the workloads, not finish them to have the PendingWorkloads status go to zero, right? can we do this test by setting the "Admission" field on the workload instead?

Reply:
basically "any" condition that is not "Assigned" will set the "Pending" to 0 , so really doesn't matter here

Reply:
hmm, it seems that we are missing to set the QW condition to `QueuedWorkloadAdmitted` when the workload gets admitted, so currently the status will continue to count the workload as pending while it is running, but this is not related to this integration test...

Reply:
yup, the k.queue.status.pending just reflects QW that have not been admitted yet , what ever happens after that is not yet reported into the k.queue.status

Reply:
that's actually the WIP of #88 , adding more things to report there

Reply:
actually the admission status is not based on the condition, it is based on the "admission" field being set, I think we need a test for that, basically have some workloads finish and some only getting the admission field set and that should impact the PendingWorkloads status

Reply:
see https://github.com/kubernetes-sigs/kueue/blob/28351208f24e5a17337bde997950209affc11f01/pkg/controller/core/queuedworkload_controller.go#L189

Reply:
> that's actually the WIP of #88 , adding more things to report there

we need to set the condition in the scheduler in this go routine: https://github.com/kubernetes-sigs/kueue/blob/28351208f24e5a17337bde997950209affc11f01/pkg/scheduler/scheduler.go#L223

Reply:
I can add that on #88 for sure